src/python/pants/backend/python/lint/autoflake/rules.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from dataclasses import dataclass
+from typing import Tuple
+
+from pants.backend.python.lint.autoflake.skip_field import SkipAutoflakeField
+from pants.backend.python.lint.autoflake.subsystem import Autoflake
+from pants.backend.python.lint.python_fmt import PythonFmtRequest
+from pants.backend.python.target_types import InterpreterConstraintsField, PythonSources
+from pants.backend.python.util_rules import pex
+from pants.backend.python.util_rules.pex import PexRequest, VenvPex, VenvPexProcess
+from pants.core.goals.fmt import FmtResult
+from pants.core.goals.lint import LintRequest, LintResult, LintResults
+from pants.core.util_rules.source_files import SourceFiles, SourceFilesRequest
+from pants.engine.fs import Digest
+from pants.engine.process import FallibleProcessResult, Process, ProcessResult
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.target import FieldSet, Target
+from pants.engine.unions import UnionRule
+from pants.util.logging import LogLevel
+from pants.util.strutil import pluralize
+
+
+@dataclass(frozen=True)
+class AutoflakeFieldSet(FieldSet):
+    required_fields = (PythonSources,)
+
+    sources: PythonSources
+    interpreter_constraints: InterpreterConstraintsField
+
+    @classmethod
+    def opt_out(cls, tgt: Target) -> bool:
+        return tgt.get(SkipAutoflakeField).value
+
+
+class AutoflakeRequest(PythonFmtRequest, LintRequest):
+    field_set_type = AutoflakeFieldSet
+
+
+@dataclass(frozen=True)
+class SetupRequest:
+    request: AutoflakeRequest
+    check_only: bool
+
+
+@dataclass(frozen=True)
+class Setup:
+    process: Process
+    original_digest: Digest
+
+
+def generate_argv(
+    source_files: SourceFiles, autoflake: Autoflake, *, check_only: bool
+) -> Tuple[str, ...]:
+    args = []
+    if check_only:
+        args.append("--check")
+    else:
+        args.append("--in-place")
+    args.append("--remove-all-unused-imports")
```

Comment:
This will likely need to be an option at some point, but fine assuming (and encouraging) side-effect-free imports for now.